tests/test_convert.py

```diff
@@ -235,6 +308,15 @@ def test_only_global_declarations():
         parse(source)
 
 
+def test_no_rebinding_in_global_scope():
+    source = """
+        include 'stdgates.inc';
+        qubit p;
+    """
+    with pytest.raises(ConversionError, match="already inserted in symbol table"):
```

Comment:
This isn't really a good error message. If these were really errors,  we would trap it and throw  a more specific error. Because trying to insert a symbol twice may arise from different operations. But since we really want to handle the messages as diagnostics, not errors, we need another way to get the same effect.